app/controllers/api/v1/users_controller.rb

```diff
@@ -3,7 +3,45 @@
 module API
   module V1
     class UsersController < API::V1::APIController
-      before_action :auth_user
+      before_action :auth_user, except: [:impersonate]
+      skip_before_action :authenticate_user!, only: [:impersonate]
+      skip_after_action :verify_authorized, only: :impersonate
+
+      def impersonate
```

Comment:
to keep it restful we could have a `API::V1::Users::ImpersonationsController#create` instead.